include/uk/init.h

```diff
@@ -126,26 +144,46 @@ typedef int (*uk_init_func_t)(struct uk_init_ctx *);
 #define uk_sys_initcall(fn)       uk_sys_initcall_prio(fn, UK_PRIO_LATEST)
 #define uk_late_initcall(fn)      uk_late_initcall_prio(fn, UK_PRIO_LATEST)
 
-extern const uk_init_func_t uk_inittab_start[];
-extern const uk_init_func_t uk_inittab_end;
+extern const struct uk_inittab_entry uk_inittab_start[];
+extern const struct uk_inittab_entry uk_inittab_end;
 
 /**
  * Helper macro for iterating over init pointer tables
  * Please note that the table may contain NULL pointer entries
  *
  * @param itr
- *   Iterator variable (uk_init_func_t *) which points to the individual
- *   table entries during iteration
+ *   Iterator variable (struct uk_inittab_entry *) which points to the
+ *   individual table entries during iteration
  * @param inittab_start
- *   Start address of table (type: const uk_init_func_t[])
+ *   Start address of table (type: const struct uk_inittab_entry[])
  * @param inittab_end
- *   End address of table (type: const uk_init_func_t)
+ *   End address of table (type: const struct uk_inittab_entry)
  */
-#define uk_inittab_foreach(itr, inittab_start, inittab_end)	\
-	for ((itr) = DECONST(uk_init_func_t*, inittab_start);	\
-	     (itr) < &(inittab_end);				\
+#define uk_inittab_foreach(itr, inittab_start, inittab_end)		\
+	for ((itr) = DECONST(struct uk_inittab_entry *, inittab_start);	\
+	     (itr) < &(inittab_end);					\
 	     (itr)++)
 
+/*
+ * Reverse iteration
+ * NOTE: The reverse iteration is intended for calling the termination functions
+ *       according to their priorities.
+ */
+#define uk_inittab_foreach_reverse(itr, inittab_start, inittab_end)	\
+	for ((itr) = (DECONST(struct uk_inittab_entry *,		\
+			 (&(inittab_end)))) - 1);			\
```

Comment:
Perhaps we could align this to the `DECONST` open paranthesis above?

Reply:
Sure 👍🏻